Scheduler/src/PIDManager.java

```diff
@@ -0,0 +1,48 @@
+import java.util.BitSet;
+
+public class PIDManager {
+    private final int minPid;
+    private final int maxPid;
+    private BitSet pidBitmap;
+
+    public PIDManager(int minPid, int maxPid) {
+        this.minPid = minPid;
+        this.maxPid = maxPid;
+        this.pidBitmap = new BitSet(maxPid - minPid + 1);
+        initializePidBitmap();
+    }
+
+    // Creates and initializes a data structure for representing pids
+    public int allocateMap() {
+        initializePidBitmap();
+        return 1;
+    }
+
+    // Allocates and returns a pid; returns -1 if unable to allocate a pid (all pids are in use)
+    public int allocatePid() {
+        int pid = pidBitmap.nextClearBit(0);
+        if (pid <= maxPid) {
+            pidBitmap.set(pid);
+            return pid + minPid;
+        } else {
+            return -1; // No available PID
+        }
+    }
+
+    // Releases a pid
+    public void releasePid(int pid) {
+        if (isValidPid(pid)) {
+            pidBitmap.clear(pid - minPid);

```

Comment:
```suggestion
            pidBitmap.clear(pid);

```

Reply:
needed because pid does not represent index only the offset